packages/fx-core/src/component/deploy/azureDeployDriver.ts

```diff
@@ -0,0 +1,250 @@
+   * @param zipBuffer zip file buffer
+   * @param config azure upload config, including azure account credential
+   * @param logger log provider
+   * @protected
+   */
+  protected static async zipDeployPackage(
+    zipDeployEndpoint: string,
+    zipBuffer: Buffer,
+    config: AzureUploadConfig,
+    logger?: LogProvider
+  ): Promise<string> {
+    let res: AxiosZipDeployResult;
+    try {
+      res = await AzureDeployDriver.AXIOS_INSTANCE.post(zipDeployEndpoint, zipBuffer, config);
+    } catch (e) {
+      throw DeployExternalApiCallError.zipDeployError(e);
+    }
+
+    if (res?.status !== 200) {
+      if (res?.status) {
+        await logger?.error(`Deployment is failed with error code: ${res.status}.`);
+      }
+      throw DeployExternalApiCallError.zipDeployError(res, res.status);
+    }
+
+    return res.headers.location;
+  }
+
+  /**
+   * loop and check azure deployment status
+   * by default, it will wait for 120 minutes
+   * @param location azure deployment location
+   * @param config azure upload config, including azure account credential
+   * @param logger log provider
+   * @protected
+   */
+  protected static async checkDeployStatus(
+    location: string,
+    config: AzureUploadConfig,
+    logger?: LogProvider
+  ): Promise<void> {
+    let res: AxiosOnlyStatusResult;
+    for (let i = 0; i < DeployConstant.DEPLOY_CHECK_RETRY_TIMES; ++i) {
+      try {
+        res = await AzureDeployDriver.AXIOS_INSTANCE.get(location, config);
+      } catch (e) {
+        throw DeployExternalApiCallError.deployStatusError(e);
+      }
+
+      if (res) {
+        if (res?.status === HttpStatusCode.HTTP_OK_ACCEPT_CODE) {
+          await waitSeconds(DeployConstant.BACKOFF_TIME_S);
+        } else if (
+          res?.status === HttpStatusCode.HTTP_OK_RESPONSE_CODE ||
+          res?.status === HttpStatusCode.HTTP_CREATE_RESPONSE_CODE
+        ) {
+          return;
+        } else {
+          if (res.status) {
+            await logger?.error(`Deployment is failed with error code: ${res.status}.`);
+          }
+          throw DeployExternalApiCallError.deployStatusError(res, res.status);
+        }
+      }
+    }
+
+    throw DeployTimeoutError.checkDeployStatusTimeout();
+  }
+
+  /**
+   * create azure zip deploy endpoint
+   * @param siteName azure app service or azure function name
+   * @protected
+   */
+  protected getZipDeployEndpoint(siteName: string): string {
+    return `https://${siteName}.scm.azurewebsites.net/api/zipdeploy?isAsync=true`;
+  }
+
+  /**
+   * create azure deploy config for Azure Function and Azure App service
+   * @param azureResource azure resource info
+   * @param azureCredential user azure credential
+   * @protected
+   */
+  protected async createAzureDeployConfig(
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredentialsBase
+  ): Promise<AzureUploadConfig> {
+    this.managementClient = new appService.WebSiteManagementClient(
```

Comment:
It seems a little bit confusing to initialize `managementClient` here. 
Since it is a data member, I guess it should be reused. But every call to `createAzureDeployConfig` creates a new instance.

Reply:
Yes, it's required by Azure Function and Azure APP Service, but Azure Storage don't need this client. So, it can't be initialized in constructor

Reply:
So maybe we can revisit the inheritance hierarchy and see whether we can make `managementClient` only known to Components that need it.
Or can we just make it a local variable, and construct it every time we need it. Drives won't be long running. Reusing an Azure Client doesn't bring much benefit.